packages/entity/src/EntityConfiguration.ts

```diff
@@ -85,6 +85,20 @@ export default class EntityConfiguration<TFields extends Record<string, any>> {
     this.dbToEntityFieldsKeyMapping = invertMap(this.entityToDBFieldsKeyMapping);
   }
 
+  private static validateSchema<TFields extends Record<string, any>>(schema: TFields): void {
+    const disallowedFieldsKeys = Object.getOwnPropertyNames(Object.getPrototypeOf({}));
+    for (const disallowedFieldsKey of disallowedFieldsKeys) {
+      // when `hasOwnProperty` is a field name, we can't call it as a method. it's still invalid though.
+      if (typeof schema.hasOwnProperty !== 'function') {
+        throw new Error(`Entity field name not allowed: hasOwnProperty`);
+      }
+
+      if (schema.hasOwnProperty(disallowedFieldsKey)) {
+        throw new Error(`Entity field name not allowed: ${disallowedFieldsKey}`);
+      }
```

Comment:
1. Good thinking to test for `hasOwnProperty` specially, but `Object.hasOwn()` takes care of this :) (Before `Object.hasOwn` was added, people would write `Object.prototype.hasOwnProperty.call(object, fieldName)` to handle the issue you identified.)
2. A more detailed error message helps specify "why" the name is disallowed.

```suggestion
      if (Object.hasOwn(schema, disallowedFieldsKey)) {
        throw new Error(`Entity field name not allowed to prevent conflicts with standard Object prototype fields: ${disallowedFieldsKey}`);
      }
```